### PR TITLE
feat: Addition of performance defaults in settings

### LINKED ISSUE #213: Performance: M4: Expose Load Generator settings in UI
**Description**
See #126 for full description of Load Generator Integration Epic. Selection of load generator should be made available in two locations:

1. as a general setting under the /settings screen. This setting will be the default for all performance tests.
1. as a per test setting under the /performance screen. This setting will be one-time for _that_ specific test.

**M4: Expose Load Generator settings in UI**
- [x] Per test selection under /performance: Update UI to select from the choice of load generators (fortio and wrk2) and include it as part of the payload to submit a job for load test, fetch response and process it the usual way. See mockup:

Screen Shot 2019-09-24 at 4 23 05 PM

- [ ] Default selection under /settings: Add load generators as a new section in the Settings screen. Fortio will be the default setting for new Meshery installations. Use the performance.svg icon provided by https://github.com/layer5io/meshery/pull/292.

performance

Meshery UI uses Material UI. Radio buttons are here - https://material-components.github.io/material-components-web-catalog/#/component/radio

**Additional context**
See [Load Generator Integration design doc](https://docs.google.com/document/d/1jZMOih3Qn1ixrq1Ve6fkdZZ4TuKNoF_qfClDZTnLNcw/edit#).

### PR DESCRIPTION
**Pull request by kushthedude**
fixes #213

### CODE REVIEW

**Comment by leecalcote** [comment]
💥

File: ui/components/MesherySettings.js [hunk 1]
```diff
@@ -121,6 +118,9 @@ class MesherySettings extends React.Component {
           case 2:
             newRoute+='#metrics'
             break;
+          case 3:
+            newRoute+='#performance'
```
Comment (leecalcote) [inline ui/components/MesherySettings.js:122]: Nice.

File: ui/components/MesherySettingsPerformanceComponent.js [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+import { withSnackbar } from 'notistack';
+import { connect } from 'react-redux';
+import CloseIcon from '@material-ui/icons/Close';
+import LoadTestTimerDialog from './load-test-timer-dialog';
+
+
+const loadGenerators = [
+  'fortio',
+  'wrk2',
+];
```
Comment (leecalcote) [inline ui/components/MesherySettingsPerformanceComponent.js:19]: We're missing nighthawk here. ;)

Comment (kushthedude) [reply ui/components/MesherySettingsPerformanceComponent.js:19]: That won't be for so long 🌟

**Comment by kushthedude** [comment]
Current Flow -:
- This PR only introduces UI Elements & Input Fields related function.
- A subsequent PR will be made which will store this preferences in-store & user will be shown an checkbox 
` Import Defaults from Settings` on the `/performance` page or if the user doesn't choose it, the values will be overridden only for a single test.

**Comment by leecalcote** [comment]
@kushthedude thanks for moving this area forward. Give thought to whether a checkbox on the /performance page is necessary, if you would. Consider that the /performance page will load with the user's defaults (whether those have been personalized or remain the project defaults) and that as the user goes to run a given test that they have the option of overriding those defaults for that specific test. 

Later, we go to provide the ability for users to create performance test profiles, users will then be enabled with a greater set of controls on what settings are used for specific test scenarios they might be running.

With respect to the /settings#performance page, given that each of the fields are marked as required, consider populating the default setting values (or the user's saved, personalized settings) in each of the three text fields on page load.

**Comment by kushthedude** [comment]
> Later, we go to provide the ability for users to create performance test profiles, users will then be enabled with a greater set of controls on what settings are used for specific test scenarios they might be running.

So what I got from your point is, when the user navigates to `/performance`, the settings defaults should populate the current values in the corresponding fields rather than a checkbox which can be overridden if the user wants to.
Seems fine to me!

**Comment by kushthedude** [comment]
@leecalcote Can we move ahead with this ?

**Comment by leecalcote** [comment]
> @leecalcote Can we move ahead with this ?

Good question. If we had multiple release channels, yes. Without additional version controls here, I'm disinclined to "release" a half-functioning feature, so to speak.

**Comment by kushthedude** [comment]
> > @leecalcote Can we move ahead with this ?
> 
> Good question. If we had multiple release channels, yes. Without additional version controls here, I'm disinclined to "release" a half-functioning feature, so to speak.

Ohk, I will push the subsequent commits here rather than another PR :+1:

**Comment by leecalcote** [comment]
@kushthedude still in draft mode here, right?

**Comment by kushthedude** [comment]
> @kushthedude still in draft mode here, right?

This still needs some work, I was testing the `saveLoadPref` endpoint which is already implemented to make sure that the functionality is not broken since Meshery is getting some new powers now-a-days.

**Comment by kushthedude** [comment]
@leecalcote The response of the GET call

```json
{c: 10, qps: 5, t: "12s", loadGenerator: "fortio"}
```

**Comment by kushthedude** [comment]
@leecalcote REady for review, please test on local!

**Pull request merged**